internal/api/courses/assignments/report_test.go

```diff
@@ -0,0 +1,75 @@
+package assignments
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/report"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+func TestCourseReport(test *testing.T) {
+	testCases := []struct {
+		email   string
+		locator string
+	}{
+		// Permission Too Low
+		{"course-other@test.edulinq.org", "-020"},
+		{"course-student@test.edulinq.org", "-020"},
+
+		// Valid Permission
+		{"course-grader@test.edulinq.org", ""},
+		{"course-admin@test.edulinq.org", ""},
+		{"course-owner@test.edulinq.org", ""},
+		{"server-admin@test.edulinq.org", ""},
+	}
+
+	for i, testCase := range testCases {
+		response := core.SendTestAPIRequestFull(test, `courses/assignments/report`, nil, nil, testCase.email)
+		if !response.Success {
+			if testCase.locator != "" {
+				if response.Locator != testCase.locator {
+					test.Errorf("Case %d: Incorrect error returned. Expected '%s', found '%s'.",
+						i, testCase.locator, response.Locator)
+				}
+			} else {
+				test.Errorf("Case %d: Response is not a success when it should be: '%v'.", i, response)
+			}
+
+			continue
+		}
+
+		if testCase.locator != "" {
+			test.Errorf("Case %d: Did not get an expected error. Expected '%s', found '%s'.",
```

Comment:
This message is off.
`response.Locator` is always "" at this point, so it shouldn't be included in the message.